spec/models/vocabulary_spec.rb

```diff
@@ -39,6 +43,130 @@
 
       it 'returns validation error' do
         expect(vocabulary.save).to be false
+        expect(vocabulary.errors.full_messages).to include 'String key can\'t be blank'
+      end
+    end
+
+    context 'when custom_field field_key is a reserved name' do
+      let(:vocabulary) do
+        FactoryBot.build(
+          :vocabulary, custom_fields: { authority: { data_type: 'string', label: 'Authority' } }
+        )
+      end
+
+      it 'returns validation error' do
+        expect(vocabulary.save).to be false
+        expect(vocabulary.errors.full_messages).to include 'Custom fields field_key cannot be a reserved field name'
```

Comment:
I suggested a change for this error message earlier in the code, so this line would have to change to:

'Custom fields authority is a reserved field name and cannot be used'